syfertext/language.py

```diff
@@ -101,54 +101,28 @@ def make_doc(self, text: Union[str, String, StringPointer]):
         are stored in a Doc object which is then returned.
         """
 
-        # TODO: huge bugs in this function. deal with the case when the function a local string `text`
-        #       after it has been already called with a remote string
-        if self.tokenizer is None:
+        if isinstance(text, StringPointer):
+            location_id = text.location.id
+        else:
+            location_id = self.owner.id
 
+        if not location_id in self.tokenizers:
             # Create the Tokenizer object
-            self.tokenizer = self.factories["tokenizer"](
+            self.tokenizers[location_id] = self.factories["tokenizer"](
                 self.vocab,
                 owner=self.owner,
                 client_id=self.owner.id,  # This is the id of the owner of the Language object using this tokenizer
             )
 
-        # If `text` is of type `StringPointer` and the pointer's `location` attribute
-        # is a worker different from the current Language object' owner, then run
-        # tokenization remotely
-        if isinstance(text, StringPointer) and text.location != self.owner:
+            if isinstance(text, StringPointer):
+                self.tokenizers[location_id] = self.tokenizers[location_id].send(text.location)
 
```

Comment:
Adding `and text.location != self.owner` guards against the case where `text` is a string pointer pointing to a string in the same local machine. I am not sure yet we might encounter this case, but just to be careful

Reply:
Good to know. I thought about this, but assumed it wouldn't be possible. I'll add this in.